src/document.jl

```diff
@@ -90,6 +90,13 @@ function parse_file(filename::AbstractString)
     XMLDocument(p)
 end
 
+function parse_file(filename::AbstractString, encoding, options::Integer)
+    p = ccall((:xmlReadFile,libxml2), Xptr, (Cstring, Ptr{Cchar}, Cint),
```

Comment:
If I wanted this to be much prettier I'd probably do a keyword-argument version with all the option enums spelled out. I could maybe have called this `read_file` to make it more clearly a direct mapping of the C API, but doesn't seem worth the trouble to have a different argument order.